test-cypress/cypress/commands/api/recording.ts

```diff
@@ -1,29 +1,85 @@
 // load the global Cypress types
 /// <reference types="cypress" />
 
-import { v1ApiPath, uploadFile, DEFAULT_DATE } from "../server";
+import { uploadFile } from "../fileUpload";
+import { v1ApiPath, DEFAULT_DATE, makeAuthorizedRequest } from "../server";
 import { logTestDescription } from "../descriptions";
 
 let lastUsedTime = DEFAULT_DATE;
 
 Cypress.Commands.add(
   "uploadRecording",
-  (cameraName: string, details: ThermalRecordingInfo, log = true) => {
+  (cameraName: string, details: ThermalRecordingInfo) => {
     const data = makeRecordingDataFromDetails(details);
 
     logTestDescription(
-      `Upload '${JSON.stringify(details)}' recording to '${cameraName}'`,
-      { camera: cameraName, requestData: data },
-      log
+      `Upload recording ${JSON.stringify(details)}  to '${cameraName}'`,
+      { camera: cameraName, requestData: data }
     );
 
     const fileName = "invalid.cptv";
     const url = v1ApiPath("recordings");
     const fileType = "application/cptv";
 
-    uploadFile(url, cameraName, fileName, fileType, data);
-    // must wait until the upload request has completed
-    cy.wait(["@addRecording"]);
+    uploadFile(url, cameraName, fileName, fileType, data, "@addRecording").then(
+      (x) => {
+        cy.wrap(x.response.body.recordingId);
+      }
+    );
+  }
+);
+
+Cypress.Commands.add(
+  "userTagRecording",
+  (recordingId: number, trackIndex: number, tagger: string, tag: string) => {
+    logTestDescription(`User '${tagger}' tags recording as '${tag}'`, {
+      recordingId,
+      trackIndex,
+      tagger,
+      tag
+    });
+
+    makeAuthorizedRequest(
+      {
+        method: "GET",
+        url: v1ApiPath(`recordings/${recordingId}/tracks`)
+      },
+      tagger
+    ).then((response) => {
+      makeAuthorizedRequest(
+        {
+          method: "POST",
+          url: v1ApiPath(
+            `recordings/${recordingId}/tracks/${response.body.tracks[trackIndex].id}/replaceTag`
+          ),
+          body: { what: tag, confidence: 0.7, automatic: false }
+        },
+        tagger
+      );
+    });
+  }
+);
+
+Cypress.Commands.add(
+  "thenUserTagAs",
+  { prevSubject: true },
+  (subject, tagger: string, tag: string) => {
+    cy.userTagRecording(subject, 0, tagger, tag);
+  }
+);
+
+Cypress.Commands.add(
+  "uploadRecordingThenUserTag",
+  (
+    camera: string,
+    details: ThermalRecordingInfo,
+    tagger: string,
+    tag: string
+  ) => {
+    cy.uploadRecording(camera, details).then((inter) => {
+      cy.log(`request is ${JSON.stringify(inter.response.body)}`);
```

Comment:
A tip you may choose to use: JSON.stringify(obj, null, "\t") will format the stringified object with indenting, which may be nicer for users to read in the console?